packages/bpk-component-page-indicator/src/BpkPageIndicator.js

```diff
@@ -0,0 +1,190 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withButtonAlignment, withRtlSupport } from 'bpk-component-icon';
+import { useSwipeable } from 'react-swipeable';
+import BpkButton from 'bpk-component-button';
+import LeftArrowIcon from 'bpk-component-icon/lg/chevron-left';
+import RightArrowIcon from 'bpk-component-icon/lg/chevron-right';
+import { isRTL, cssModules } from 'bpk-react-utils';
+
+import STYLES from './BpkPageIndicator.module.scss';
+
+const getClassName = cssModules(STYLES);
+const AlignedLeftArrowIcon = withButtonAlignment(withRtlSupport(LeftArrowIcon));
+const AlignedRightArrowIcon = withButtonAlignment(
+  withRtlSupport(RightArrowIcon),
+);
+const DISPLAYED_TOTAL = 5;
+const START_SCROLL_INDEX = Math.floor(DISPLAYED_TOTAL / 2);
+
+export type Props = {
+  className: ?string,
+  showNav: ?boolean,
+  currentItem: number,
+  totalItems: number,
+  updateItemCallback: ?() => void,
+  itemClassName: ?string,
+  activeItemClassName: ?string,
+  ariaLabel: ?string,
+};
+
+const KEYS = {
+  ENTER: 'Enter',
+  SPACE: 'Spacebar',
+};
+
+const BpkPageIndicator = (props: Props) => {
+  const {
+    activeItemClassName,
+    ariaLabel,
+    className,
+    currentItem,
+    itemClassName,
+    showNav,
+    totalItems,
+    updateItemCallback,
+    ...rest
+  } = props;
+  const classNames = getClassName('bpk-page-indicator', className);
+
+  const updateItem = (index) => {
+    let newItem = index;
+    if (index < 0) {
+      newItem = totalItems - 1;
+    }
+    if (index >= totalItems) {
+      newItem = 0;
+    }
+    updateItemCallback(newItem);
+  };
+
+  const handleKeyboardEvent = (e, index) => {
+    if (e.key === KEYS.ENTER || e.key === KEYS.SPACE || e.key === ' ') {
+      e.preventDefault();
+      updateItemCallback(index);
+    }
+  };
+
+  const direction = isRTL() ? 1 : -1;
+  const handlers = useSwipeable({
+    onSwipedLeft: () => updateItem(currentItem - direction),
+    onSwipedRight: () => updateItem(currentItem + direction),
+    trackMouse: true,
+    trackTouch: true,
+    preventScrollOnSwipe: true,
+  });
+
+  const bulletArray = [];
+  for (let i = 0; i < totalItems; i += 1) {
+    bulletArray.push(i);
+  }
+
+  return (
+    // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
+    <div {...handlers} className={classNames} {...rest}>
+      {showNav && (
+        <BpkButton
+          iconOnly
+          link
+          onClick={() => {
+            updateItem(currentItem - 1);
+          }}
+        >
+          <AlignedLeftArrowIcon />
+          <span className="visually-hidden">Prev</span>
```

Comment:
Seems we can just set the `aria-label` on the button instead of this span here, eg: `aria-label="Previous Slide"`

Reply:
And seems that it's easy to put the prev and next button together, we can just use css to make them `space-between` for bullets

Reply:
We can also consider wrapping a component `NavButton` for the prev and next buttons

Reply:
The button component needs a `disabled` property to ensure it cannot be left clickable at `currentIndex=0`, the same for right click event

Reply:
I try to wrap them into the same NavButton, but could not find a good way to overlap this component and page indicator. This is what it looks like after I have combine the `NavButton` and set the position of the page indicator to `absolute` and added a partial margin. It looks like the alignment is not as standard as with flex+center.

Reply:
Have extract a `NavButton` component to wrap the nav button. Thank you for your comment!